src/components/WorldMap.js

```diff
@@ -113,18 +113,18 @@ export class WorldMap extends Component {
     this.setState({
       map
     });
-
-    if (localStorage.getItem('geolocation')) {
-      const [lat, long] = getCoords();
-      map.setView([lat, long], 5);
-    }
   }
 
   componentWillUnmount() {
     this.state.map.remove();
   }
 
-  render() {
+  render(_, { map }) {
```

Comment:
this will reset the map each re-render, right? is there something like an "updated" callback in preact we can use when the property changes?

Reply:
the map should only rerender when `countriesData` changes. Other 'updates' to the map are handled/mutated on the ref iirc

Im a bit confused though, because in `MainPage` the `location` state changes, which means it should pass the new prop to `WorldMap`, but it doesnt seem to trigger a rerender.

Reply:
let's  keep it like this for now